crm_claim_rma_number/model/crm_claim_rma.py

```diff
@@ -0,0 +1,125 @@
+###############################################################################
+
+from openerp import models, fields, api
+
+
+class claim_line(models.Model):
+
+    _inherit = 'claim.line'
+
+    @api.model
+    def _get_sequence_number(self):
+        """
+        @return the value of the secuence for the number field in the
+        claim.line model.
+        """
+        seq_obj = self.env['ir.sequence']
+        return seq_obj.get('claim.line')
+
+    number = fields.Char(
+        readonly=True,
+        default='/',
+        help='Claim Line Identification Number')
+
+    # Field "number" is assigned by default with "/"
+    # then this constraint ever is broken
+    # _sql_constraints = [
+    #     ('number_uniq', 'unique(number, company_id)',
+    #         'Internal RMA number must be unique per Company!'),
+    # ]
+
+    @api.model
+    def create(self, vals):
+        """
+        @return wirte the identify number once the claim line is create.
+        """
+        vals = vals or {}
+        if ('number' not in vals) or (vals.get('number', False) == '/'):
+            vals['number'] = self._get_sequence_number()
+        res = super(claim_line, self).create(vals)
+        return res
+
+    @api.multi
+    def name_get(self):
+        result = []
+        for cl in self:
+            name = "%s - %s" % (cl.claim_id.number, cl.name)
+            result.append((cl.id, name))
+        return result
+
+
+class crm_claim_type(models.Model):
+
+    _inherit = 'crm.claim.type'
+
+    ir_sequence_id = \
+        fields.Many2one('ir.sequence',
+                        string='Sequence Number',
+                        default=lambda self:
+                        self.env['ir.sequence'].
+                        search([('code', '=', 'crm.claim.rma.basic')])
+                        )
+
+
+class crm_claim(models.Model):
+
+    _inherit = 'crm.claim'
+
+    rma_number = fields.Char('RMA Number', size=128,
+                             help='RMA Number provided by supplier')
+
+    def _get_sequence_number(self, cr, uid, code_id, context=None):
+        seq_obj = self.pool.get('ir.sequence')
+        res = '/'
+        claim_type_obj = self.pool.get('crm.claim.type')
+        claim_type = claim_type_obj.browse(cr, uid, code_id)
+        code = claim_type.ir_sequence_id.code
+        if code:
+            res = seq_obj.get(cr, uid, code) or '/'
+        return res
+
+    @api.v7
+    def create(self, cur, uid, vals, context=None):
+        if ('number' not in vals) or (vals.get('number') == '/') \
+                or vals.get('number') is False:
+            vals['number'] = self._get_sequence_number(cur, uid,
+                                                       vals['claim_type'],
+                                                       context=context)
+        new_id = super(crm_claim, self).create(cur, uid, vals, context=context)
+        return new_id
+
+    @api.v7
+    def copy_data(self, cr, uid,
```

Comment:
2015-08-05 15:38 GMT-05:00 Pedro M. Baeza <notifications@github.com>:

> Why is this done this way? Use copy=False in the fields definition for
> avoiding these fields to be copied
>

+1

(remember this modules were migrated from 7 just waiting to make as les
changes as possible, but due to this work we will improve it.

@yaniaular @olvalr proceed please when  you can.


-- 
--------------------
Saludos Cordiales

CEO at Vauxoo <https://www.vauxoo.com> Odoo's Gold Partner.

[image: --]
Nhomar Hernandez
[image: http://]about.me/nhomar
<http://about.me/nhomar?promo=email_sig>